packages/core/content-type-builder/server/src/controllers/validation/types.ts

```diff
@@ -219,24 +219,8 @@ const getTypeShape = (
       return {
         required: validators.required,
         repeatable: yup.boolean(),
-        component: yup
-          .string()
-          .test({
-            name: 'Check max component nesting is 1 lvl',
-            test(compoUID: unknown) {
-              const targetCompo = strapi.components[compoUID as UID.Component];
-              if (!targetCompo) return true; // ignore this error as it will fail beforehand
-
-              if (modelType === modelTypes.COMPONENT && hasComponent(targetCompo)) {
-                return this.createError({
-                  path: this.path,
-                  message: `${targetCompo.modelName} already is a nested component. You cannot have more than one level of nesting inside your components.`,
-                });
-              }
-              return true;
-            },
-          })
-          .required(),
+        // TODO: Add correct server validation for nested components
+        component: yup.string().required(),
```

Comment:
Maybe you had plans to do this but I think if we're saying that the max level of nesting is 6 that should be enforced on the frontend and backend. wdyt?

Reply:
we actually never planned to enforce on the backend, and whatever was previously here was not working from all places anyways, so yeah would leave it to a different point in time since this will just add more to the complexity and  no one should be hitting this point anyways

Reply:
I still think the validation here should match our intentions for the feature
and prevent incorrect setups being made through the backend. Could we somehow
count the deepest level the component is being used at in a util & use it
similarly to `hasComponent` ?

wdyt? @innerdvations

Reply:
What would it take to add such validation, is it only about checking the number of nested components in the received data?

Anyway, we should try to have the same rules in the admin and server. So I would add this check here too if it's possible and doesn't deviate from the original PR goal.

Reply:
will do that at a second iteration, we would be adding the limitation rules to the server as well
but the goal here was to add the limitations only on the frontend considering the server already accepted any amount of nested components, and too also move with quickly.
we would basically have to duplicate the same depth counting algorithm on the frontend to add the rule, it's a little complex but should get the job done